standard_hostpool.go

```diff
@@ -119,6 +126,7 @@ func NewWithOptions(hosts []string, options StandardHostPoolOptions) HostPool {
 		if p.maxFailures > 0 {
 			// We test for failures > maxFailures, so need an extra slot in the buffer.
 			e.failures = NewRingBuffer(p.maxFailures + 1)
+			e.successes = NewRingBuffer(p.maxFailures + 1)
```

Comment:
why is the size of the successes ring buffer `p.maxFailures + 1`? won't this mean we can only observe `p.maxFailures + 1` many successes?